File: Source/Presentation/Web/ClientApp/src/lib/charts/model/line-chart.ts
export interface ChartDimensions
{
    width: number;
    height: number;
}

export interface LinechartAxisLabel
{
    isVisible: boolean;
    text: string;
}

export interface LinechartAxis
{
    isVisible: boolean;
    label: LinechartAxisLabel;
}

export interface LinechartColorScheme
{
    domain: string[];
}

export interface LinechartDataSet<TX, TY>
{
    name: string;
    series: LinechartDataRow<TX, TY>[];
}

export interface LinechartDataRow<TX, TY>
{
    name: TX;
    value: TY;
}

export interface LinechartConfiguration<TX, TY>
{
    dimensions: ChartDimensions;
    axisX: LinechartAxis;
    axisY: LinechartAxis;
    colorScheme: LinechartColorScheme;
    autoScale: boolean;
    showLegend: boolean;
    withGradient: boolean;
    data: LinechartDataSet<TX, TY>[];
}

File: Source/Presentation/Linechart.Generics/Linechart/Model/Linechart.cs
using Presentation.Linechart.Design;

namespace Presentation.Linechart.Generics.Linechart.Model
{
    public class Linechart<TX, TY> : ILinechartConfiguration<TX, TY>
    {
        public IChartDimensions Dimensions { get; set; }
        public ILinechartAxis AxisX { get; set; }
        public ILinechartAxis AxisY { get; set; }
        public ILinechartColorScheme ColorScheme { get; set; }
        public ILinechartDataSet<TX, TY>[] Data { get; set; }
        public bool AutoScale { get; set; }
        public bool ShowLegend { get; set; }
        public bool WithGradient { get; set; }
    }
}